jax/interpreters/xla.py

```diff
@@ -463,22 +464,88 @@ def jaxpr_collectives(jaxpr):
 
 ### xla_call underlying jit
 
-def _xla_call_impl(fun: lu.WrappedFun, *args, device, backend, name):
-  compiled_fun = _xla_callable(fun, device, backend, name, *map(arg_spec, args))
+def _xla_call_impl(
+    fun: lu.WrappedFun,
+    *args,
+    device: Optional[xe.Device],
```

Comment:
xe.Device failed mypy for me. I have been using just 'Device' defined at top of the file.

Reply:
Will address all these type issues once the comments are in from other folks. FYI the tests all pass internally (with pytype checking the types).

Reply:
I need to add a `.pyi` file for the XLA extension I think. I'll get on it soonish.

Reply:
I've dropped these hints for now based on other comments re arg formatting.